### PR TITLE
Audit and Fix data-conversions.md

### LINKED ISSUE #123: Audit and fix Resource-Appendecies/data-conversions.md
- Look for any typos
- Make sure all Markdown is correctly done.  The conversion was done automatically by a tool, so there may be more idiomatic ways to do some things
- Check the page in a Desktop browser.  Does it look OK?  Are there any issues that need to be fixed?
- Check the page in a Mobile browser.  Does it look OK?  Are there any issues that need to be fixed?
- Try the page in both Light and Dark mode.  Are there any issues that need to be fixed?
- Try running the page through Lighthouse: https://developers.google.com/web/tools/lighthouse.  Are there any issues that need to be fixed?
- Try running the page through Web Hint: https://webhint.io/.  Are there any issues that need to be fixed?
- Tables that are really wide (types.md) could be converted to use SVG or flex/grid. Make wide visual tables fit in the available viewport.  Another example is the table in `docs/D-Modularity/pointers.md`, where we should reduce the vertical padding/margins on tables. The tables are too large, and probably shouldn't be tables at all.
- Prefer Markdown tables vs. raw HTML tags
- Fix colours in tables so it uses global styles vs. inline styles
- Use alt tags for all images.  See https://supercooldesign.co.uk/blog/how-to-write-good-alt-text for suggestions
- Avoid blockquote for indentation, and prefer q or blockquote only when actually quoting.  Use Admonitions instead. Example: `docs/D-Modulatirty/pointers.md`: there are places where > Blockquote are being used and Admonitions could replace it. “Note:” Lightbulb info Admonition.
- Make sure images work in dark and light modes.  For example, `docs/8-Computations/logic.md` Flags image is white background, needs to work in dark.  Create a “negative” or SVG so we can do Dark Mode?  Can we programmatically create this negative?
- Change colours for tables (don’t use Yellow).  We should use the default colour theme.
- Avoid using `<b>` and `<i>` to do emphasis.
- Use backticks and all code for memory addresses and such in tables that use things like void, int, function names, memory addresses.
- Make sure all code blocks use syntax highlighting
- centre images horizontally (esp. when they are small) so they work well at all responsive sizes
- check sub-script and super-script that they are correctly used in all cases
- consider including inter-site links to other pages that have related info
- all links should have descriptive text for accessibility
- Make sure that sub-headings (right-hand side) for each document are correct.
- Improve readability of code comments (i.e., `//` and `/*...*/`) in dark mode (even light mode might need some work)
- In places where `***...***` was used as a fourth heading level, use `####` instead.
- Fix Frontmatter for page to include proper id, title, slug, etc.

### PR DESCRIPTION
**Pull request by joshuali7536**
Fixes #123

- Checked for typos and grammar errors
- Checked page in desktop, mobile, light & dark
- Checked using Lighthouse & Web Hint
- Added alt tags to images
- Added Frontmatter

Note:
The tables are kept as HTML tags due to formatting requirements. Markdown tables cannot have merged cells in a row and cannot have no header row. Table formatting as seen below:
image

### CODE REVIEW

**Comment by Kevan-Y** [comment]
@joshuali7536 Hi, there is a recent change where we remove slug (#142). Could you kindly remove slug in all you changed files, thank you

**Comment by joshuali7536** [comment]
@Kevan-Y the `slug` has been removed from the modified files.

File: docs/Resources-Appendices/data-conversions.md [hunk 1]
```diff
@@ -19,7 +21,7 @@ This chapter describes how to convert across binary, hexadecimal and decimal rep
 
 ## Binary - Hexadecimal
 
-The most convenient base for storing byte-wise information is hexadecimal (base 16). Two hexadecimal (base 16) digits can represent one byte of information. Each hexadecimal digit represents 4 bits of binary information.
+The most convenient base for storing byte-wise information is hexadecimal (base 16). Two hexadecimal digits can represent one byte of information. Each hexadecimal digit represents 4 bits of binary information.
```
Comment (Kevan-Y) [inline docs/Resources-Appendices/data-conversions.md:24]: Why in `Two hexadecimal` `(base 16)`is removed?

File: docs/Resources-Appendices/data-conversions.md [hunk 2]
```diff
@@ -1,4 +1,6 @@
 ---
+id: data-conversions
+title: Data Conversions
 sidebar_position: 3
 ---
```
Comment (Kevan-Y) [inline docs/Resources-Appendices/data-conversions.md:5]: Add Description

**Review by Kevan-Y — CHANGES_REQUESTED** [review]

File: docs/Resources-Appendices/data-conversions.md (hunk 1 above)
Comment (joshuali7536) [reply docs/Resources-Appendices/data-conversions.md:24]: Oh I thought it wouldn't need to be repeated, I'll add it back.

**Review by Kevan-Y — APPROVED** [review]

**Comment by joshuali7536** [comment]
Branch has been rebased

**Review by CameronGray1210 — APPROVED** [review]
Great - good work!

**Pull request merged**